### PR TITLE
Update and add missing docstrings related to ML functionalities

### PR DESCRIPTION
**Pull request by LukasBeiske**

### CODE REVIEW

**Comment by Tobychev** [comment]
Looks like your doc-tests got hit by the scipy devs letting their ssl cert expire :(

**Review by Tobychev — APPROVED** [review]

**Comment by maxnoe** [comment]
> Looks like your doc-tests got hit by the scipy devs letting their ssl cert expire :(

Fixed now, restarted the CI

File: ctapipe/reco/sklearn.py [hunk 1]
```diff
@@ -437,6 +436,9 @@ class ParticleClassifier(SKLearnClassificationReconstructor):
     property = ReconstructionProperty.PARTICLE_TYPE
 
     def __call__(self, event: ArrayEventContainer) -> None:
```
Comment (maxnoe) [inline ctapipe/reco/sklearn.py:438]: subclasses should inherit docstrings from their parents automatically in sphinx I think. Was this needed?

Comment (LukasBeiske) [reply ctapipe/reco/sklearn.py:438]: Oh yes, you are right. I was confused since it got overwritten everywhere else, but I think it would be better to just use the docstring from the parent class for `__call__` and `predict_table` in all subclasses here. 👍

File: ctapipe/reco/sklearn.py [hunk 2]
```diff
@@ -292,7 +295,7 @@ class SKLearnClassificationReconstructor(SKLearnReconstructor):
     ).tag(config=True)
 
     invalid_class = traits.Integer(
-        default_value=-1, help="The label to fill in case no prediction could be made"
+        default_value=-1, help="The label to fill in case no prediction could be made."
```
Comment (Tobychev) [inline ctapipe/reco/sklearn.py:298]: To me "label" sounds very string-like, might be better to say "value" when talking about an integer.

Comment (LukasBeiske) [reply ctapipe/reco/sklearn.py:298]: How about "label value" like for `positive_class`?

Comment (maxnoe) [reply ctapipe/reco/sklearn.py:298]: > To me "label" sounds very string-like

This is the "label" in the machine learning sense: i.e. the true / target quantity.

Comment (Tobychev) [reply ctapipe/reco/sklearn.py:298]: Label value sounds fine @LukasBeiske

**Review by Tobychev — APPROVED** [review]

**Review by maxnoe — APPROVED** [review]
